ansible-tests/tripleo-ansible-inventory.py

```diff
@@ -0,0 +1,178 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# TODO
+# Make it possible to specify which plan to generate inventory for
+# Get info from ironic for hosts prior to deployment
+# Get info from nova for more accurate addresses
+# Only add host if it is up according to nova
+
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import re
+import sys
+
+from oslo_config import cfg
+
+opts = [
+    cfg.StrOpt('host', help='List details about the specific host'),
+    cfg.BoolOpt('list', help='List active hosts'),
+    cfg.StrOpt('username'),
+    cfg.StrOpt('password'),
+    cfg.StrOpt('auth-url'),
+    cfg.StrOpt('project-name'),
+    cfg.StrOpt('group-regex'),
+]
+
+try:
+    from heatclient.v1 import client as heat_client
+except ImportError:
+    print('heatclient is required',file=sys.stderr)
+    sys.exit(1)
+try:
+    from keystoneclient.v3 import client as keystone_client
+except ImportError:
+    print('keystoneclient is required', file=sys.stderr)
+    sys.exit(1)
+
+
+def _parse_config():
+    default_config = os.environ.get('TRIPLEO_INVENTORY_CONFIG')
+    if default_config:
+        default_config = [default_config]
+
+    configs = cfg.ConfigOpts()
+    configs.register_cli_opts(opts)
+    configs(prog='tripleo-ansible-inventory',
+            default_config_files=default_config)
+    if configs.auth_url is None:
+        if "OS_AUTH_URL" in os.environ:
+            configs.auth_url = os.environ.get('OS_AUTH_URL')
+        else:
+            print('ERROR: auth-url not defined and OS_AUTH_URL environment variable missing, unable to proceed.', file=sys.stderr)
+            sys.exit(1)
+    if configs.username is None:
+        if "OS_USERNAME" in os.environ:
+            configs.username = os.environ.get('OS_USERNAME')
+    if configs.password is None:
+        if "OS_PASSWORD" in os.environ:
+            configs.password = os.environ.get('OS_PASSWORD')
+    if configs.project_name is None:
+        if "OS_TENANT_NAME" in os.environ:
+            configs.project_name = os.environ.get('OS_TENANT_NAME')
+    if '/v2.0' in configs.auth_url:
+        configs.auth_url = configs.auth_url.replace('/v2.0', '/v3')
+    return configs
+
+
+class HeatInventory(object):
+    def __init__(self, configs):
+        self.configs = configs
+        self._ksclient = None
+        self._hclient = None
+
+
+    def list(self):
+        ret = {'undercloud': {
+               'hosts': ['localhost'],
+                'vars': {
+                    'ansible_connection': 'local'
+                },
+            }
+        }
+        heatclient = self.hclient
+        controller_id = heatclient.resources.get('overcloud', 'Controller').physical_resource_id
```

Comment:
Can we wrap this (and the compute call below as well) in a try/except block? It should work even when the overcloud isn't deployed yet (e.g. for our undercloud validations).